src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -52,85 +67,75 @@ private static string[] ParseMultiValue(string value)
                 tempStringCollection.Add(singleValue.Trim());
             }
 
-            string[] stringArray = new string[tempStringCollection.Count];
-            tempStringCollection.CopyTo(stringArray, 0);
-            return stringArray;
+            return tempStringCollection.ToArray();
         }
 
-        static HeaderInfoTable()
+        private static readonly Dictionary<string, HeaderInfo> s_headerHashTable = new Dictionary<string, HeaderInfo>()
```

Comment:
Doesn't this need the same comparer that was previously being passed to the Hashtable for doing case-insensitive ASCII comparisons?